### ISSUE TITLE
Hyperlink annotations regression

### ISSUE THREAD

**Issue by GitHubRulesOK** [sumatrapdfreader/sumatrapdf#1914]
Keeping this as a separate issue from other hyperlink regressions as seems to be a different cause
From Forum https://forum.sumatrapdfreader.org/t/links-inside-a-pdf-not-working/3783
[UNE-EN_1991.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/6216075/UNE-EN_1991.pdf)

Page 15 has a standard "GoTo" page link to 49 which works, however on page 49 there are stamped ? arrows that seem to be annotation links back to the source page thus the first "return" (volver) arrow should link back to page 15.

This works most (but not all) times whilst viewing in a browser , and Did work in SumatraPDF Version 3.1.2 but not in 3.2 or latest. (they do not work in MuPDF 1.18)

**Comment by kjk** [comment]
In 3.1.2 in `pdf_load_link` there is not check for `Subtype` so it accepts `Widget` as link, in addition to `Link`

I've opened mupdf bug: https://bugs.ghostscript.com/show_bug.cgi?id=703834

This would fix it:
```
diff --git a/mupdf/source/pdf/pdf-link.c b/mupdf/source/pdf/pdf-link.c
index 5ccefcf1a..8de33b464 100644
--- a/mupdf/source/pdf/pdf-link.c
+++ b/mupdf/source/pdf/pdf-link.c
@@ -405,7 +405,7 @@ pdf_load_link(fz_context *ctx, pdf_document *doc, pdf_obj *dict, int pagenum, fz
        fz_link *link = NULL;

        obj = pdf_dict_get(ctx, dict, PDF_NAME(Subtype));
-       if (!pdf_name_eq(ctx, obj, PDF_NAME(Link)))
+       if (!(pdf_name_eq(ctx, obj, PDF_NAME(Link)) || pdf_name_eq(ctx, obj, PDF_NAME(Widget))))
                return NULL;

        obj = pdf_dict_get(ctx, dict, PDF_NAME(Rect));
```

**Comment by kjk** [comment]
mupdf closed this as wontfix with this explanation:
```
Those are buttons, not links :)

Older versions of MuPDF didn't handle buttons, so we converted simple buttons with follow-a-link actions into links. We can't do that anymore without risking breaking interactive documents using javascript actions in forms and widgets.
```

I guess this means this should be handled as a button in JavaScript

**Comment by GitHubRulesOK** [comment]
Since I agree SumatraPDF should not currently be using Scripting as exploitable I am closing as cant fix

**Comment by GitHubRulesOK** [comment]
> mupdf closed this as wontfix with this explanation:
> 
> ```
> Those are buttons, not links :)
> 
> Older versions of MuPDF didn't handle buttons, so we converted simple buttons with follow-a-link actions into links. We can't do that anymore without risking breaking interactive documents using javascript actions in forms and widgets.
> ```
> 
> I guess this means this should be handled as a button in JavaScript

I am not so certain that JS or ecma script is required as they are visually more like /Action script variant of goto

339 0 obj
<</A 381 0 R/AP<</D 385 0 R/N 386 0 R>>/H/P/MK<</I 382 0 R/IF<</FB true>>/TP 1>>/P 133 0 R/Parent 345 0 R/Rect[19.2366 521.185 45.0367 542.385]/Subtype/Widget/Type/Annot>>
endobj

**Comment by tobidelbruck** [comment]
Thanks, comments were very helpful. The root of the problem is not muPDF but rather pdfcomment. See [2391](https://github.com/sumatrapdfreader/sumatrapdf/issues/2391#issuecomment-1723504101). .

**Comment by u-fischer** [comment]
@tobidelbruck  No pdfcomment is not the problem. It is not a bug that the tooltip is implemented as a button field. If sumatra or mupdf do not support such button fields (or other fields like checkbox or radiobuttons or other PDF features), it is their decision.  

Note that while fields can take javascript (ecma-script) actions, this is not required and unless you want to calculate a field value you normally do not need it. Lots of document with form fields do not use javascript and the pdfcomment package doesn't add any javascript code.

**Comment by tobidelbruck** [comment]
Agree. Still muPDF does not show \pdftooltip tooltips. As a workaround, see https://github.com/tobidelbruck/acronyms-with-tooltips-latex . Both annotation types are drawn. This helps for some viewers, e.g. e.g. muPDF.

**Comment by GitHubRulesOK** [comment]
Updated summary that such links do NOT work in the online viewer and do not show as links in either SumatraPDF other viewers but back will return to the page they jumped from it is only Desktop Acrobat **shows the tip to "volver**" by press as if a JS button HOWEVER they DO work in Edge Powered by Acrobat even without the tooltip.

<img width="644" height="275" alt="Image" src="https://github.com/user-attachments/assets/350f919e-7e67-4987-9424-a04a06b3bcb4" />

<img width="633" height="287" alt="Image" src="https://github.com/user-attachments/assets/7ba3e019-64eb-49f0-8093-b47515afc83c" />

<img width="700" height="400" alt="Image" src="https://github.com/user-attachments/assets/42c282bd-bbea-49c4-9d69-02107d1b3219" />

<img width="771" height="392" alt="Image" src="https://github.com/user-attachments/assets/a992caa2-aef4-4db1-a2c0-fe00b1529061" />

**Comment by GitHubRulesOK** [comment]
@kjk that is as good as Edge **the links show IF links highlight is on** however like Edge the tooltip is not seen

<img width="485" height="337" alt="Image" src="https://github.com/user-attachments/assets/bbf1e58a-c254-4039-a845-e28d385f5c6e" />
